### ISSUE TITLE
Chef Zero provisioner does not respect `require_chef_omnibus` config

### ISSUE THREAD

**Issue by schisamo** [test-kitchen/test-kitchen#243]
This was originally reported as part of https://github.com/opscode/test-kitchen/issues/240.
### TL;DR

The Chef Zero provisioner [instantiates an instance of `Chef::Application::Client`](https://github.com/opscode/test-kitchen/blob/master/support/chef-client-zero.rb#L63-L67) when running `chef-client` that does not match the instance's Omnibus Chef install. This means that if a user declares Chef 11.4.4 via their `.kitchen.yml` like so:

```
driver_config:
  require_chef_omnibus: 11.4.4
```

the actual `chef-client` run will use the latest version of Chef (currently 11.8.0).
### Longer explanation of issue

In the past when we installed `knife-essentials` and `chef-zero` into the Omnibus Chef's `GEM_PATH` that version of Chef was used for the `chef-client` run. Now that we [sandbox the knife-essentials and chef-zero install into an isolated `GEM_PATH`](https://github.com/opscode/test-kitchen/blob/master/lib/kitchen/provisioner/chef_zero.rb#L46) the latest version of Chef is also [installed via a knife-essentials dependancy](https://github.com/jkeiser/knife-essentials/blob/master/knife-essentials.gemspec#L21). It's probably easiest to just render a client.rb file and shell-out and run `/opt/chef/bin/chef-client` directly.
